co-simulation/fed/mosaic-infrastructure/src/main/java/org/eclipse/mosaic/fed/infrastructure/ambassador/V2xRegistrationReceiver.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2023 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.eclipse.mosaic.fed.infrastructure.ambassador;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.SocketException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Queue;
+import com.google.gson.Gson;
+
+/**
+ * Worker thread Runnable for operating a listen socket to receive outbound V2X Messages from Infrastructure Device instances
+ * This {@link Runnable} instance will operate a UDP socket to subscribe to packets from the V2x infrastructure's
+ * adapter. Upon receiving a packet, it will be enqueued for the primary thread to process the data once it ticks to a
+ * simulation processing step
+ * NOTE: TODO See carma.ambassador for reference
+ */
+public class V2xRegistrationReceiver implements Runnable {
+    private Queue<V2xRegistrationMessage> rxQueue = new LinkedList<>();
+    private DatagramSocket listenSocket = null;
+    private static final int listenPort = 1515; // TODO
+    private boolean running = true;
```

Comment:
Probably should initialize this as false, make it true when calling run successful and make it false again on error handling or stop calls right.